packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -772,6 +803,63 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         return iconClass.concat(additionalClasses).join(' ');
     }
 
+    /**
+     * Add the parent id of the nodes that need tree indent guideline into a set
+     * @param node the tree node.
+     */
+    protected addActiveNodeParent(node: TreeNode): void {
+        if (this.isExpandable(node)) {
+            if (node.children && node.children.length > 0) {
+                this.parentOfActiveIndentGuideline.add(node.id);
+            }
+        } else {
+            const parent = node.parent;
+            if (parent && parent.children.length > 0) {
```

Comment:
I mean, if we get a node and fetch its parent, wouldn't `parent.children` be at least 1, always?